packages/editor-ui/src/components/ExecutionsUsage.vue

```diff
@@ -0,0 +1,157 @@
+						{{
+							locale.baseText('executionUsage.currentUsage.count', {
+								interpolate: { days: daysLeftOnTrial.toString() },
+							})
+						}}
+					</n8n-text>
+				</template>
+			</i18n>
+		</div>
+		<div v-if="isTrialExpired" :class="$style.usageText">
+			<n8n-text size="xsmall" color="danger">
+				{{ locale.baseText('executionUsage.expired.text') }}
+			</n8n-text>
+		</div>
+		<div v-if="!trialHasExecutionsLeft" :class="$style.usageText">
+			<n8n-text size="xsmall">
+				{{ locale.baseText('executionUsage.ranOutOfExecutions.text') }}
+			</n8n-text>
+		</div>
+		<div v-if="!isTrialExpired" :class="$style.usageCounter">
+			<div>
+				<progress
+					:class="[
+						trialHasExecutionsLeft ? $style.progressBarSuccess : $style.progressBarDanger,
+						$style.progressBar,
+					]"
+					:value="currentExecutions"
+					:max="maxExecutions"
+				></progress>
+			</div>
+			<div :class="$style.executionsCountSection">
+				<n8n-text size="xsmall" :color="trialHasExecutionsLeft ? 'text-dark' : 'danger'">
+					{{ currentExecutions }}/{{ maxExecutions }}
+				</n8n-text>
+				<n8n-text size="xsmall" :color="trialHasExecutionsLeft ? 'text-dark' : 'danger'">{{
+					locale.baseText('executionUsage.label.executions')
+				}}</n8n-text>
+			</div>
+		</div>
+
+		<div :class="$style.upgradeButtonSection">
+			<n8n-button
+				:label="locale.baseText('executionUsage.button.upgrade')"
+				size="mini"
+				icon="gem"
+				type="success"
+				:block="true"
+				@click="onUpgradeClicked"
+			/>
+		</div>
+	</div>
+</template>
+
+<script setup lang="ts">
+import { i18n as locale } from '@/plugins/i18n';
+import { DateTime } from 'luxon';
+import type { Cloud } from '@/Interface';
+import { useSettingsStore } from '@/stores';
+import { CHANGE_PLAN_PAGE_PRODUCTION, CHANGE_PLAN_PAGE_STAGING } from '@/constants';
+import { computed } from 'vue';
+
+type CloudPlanData = Cloud.PlanData & { metadata: Cloud.PlanMetadata; usage: Cloud.PlanUsage };
+
+const props = defineProps<{ cloudPlanData: CloudPlanData }>();
+
+const now = DateTime.utc();
+
+const settingsStore = useSettingsStore();
+
+const daysLeftOnTrial = computed(() => {
+	const { days = 0 } = getPlanExpirationDate().diff(now, ['days']).toObject();
+	return Math.ceil(days);
+});
+
+const isTrialExpired = computed(() => {
+	const trialEndsAt = DateTime.fromISO(props.cloudPlanData.expirationDate);
+	return now.toMillis() > trialEndsAt.toMillis();
+});
+
+const getPlanExpirationDate = () => DateTime.fromISO(props.cloudPlanData.expirationDate);
+
+const trialHasExecutionsLeft = computed(
+	() => props.cloudPlanData.usage.executions < props.cloudPlanData.monthlyExecutionsLimit,
+);
+
+const currentExecutions = computed(() => props.cloudPlanData.usage.executions);
+
+const maxExecutions = computed(() => props.cloudPlanData.monthlyExecutionsLimit);
+
+const onUpgradeClicked = () => {
+	location.href =
+		settingsStore.settings.license.environment === 'production'
```

Comment:
just highlighting this does not work yet.. an okay trick here might be just to check if host has `-stage` in it.. 😅 or maybe we can add the env to `n8nMetadata`

Reply:
in this case it probably does not event matter.. we can simply point to production.. but when we fetch the data for users, we need to figure this out.

Reply:
This one only point to production